test/position-managers/Execute.t.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
+import {PoolManager} from "@uniswap/v4-core/src/PoolManager.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {Deployers} from "@uniswap/v4-core/test/utils/Deployers.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {BalanceDelta} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
+import {PoolSwapTest} from "@uniswap/v4-core/src/test/PoolSwapTest.sol";
+import {LiquidityAmounts} from "@uniswap/v4-core/test/utils/LiquidityAmounts.sol";
+import {TickMath} from "@uniswap/v4-core/src/libraries/TickMath.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+import {SafeCast} from "@uniswap/v4-core/src/libraries/SafeCast.sol";
+
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+import {IPositionManager, Actions} from "../../src/interfaces/IPositionManager.sol";
+import {PositionManager} from "../../src/PositionManager.sol";
+import {LiquidityRange, LiquidityRangeId, LiquidityRangeIdLibrary} from "../../src/types/LiquidityRange.sol";
+
+import {LiquidityFuzzers} from "../shared/fuzz/LiquidityFuzzers.sol";
+
+import {LiquidityOperations} from "../shared/LiquidityOperations.sol";
+import {Planner} from "../utils/Planner.sol";
+
+contract ExecuteTest is Test, Deployers, GasSnapshot, LiquidityFuzzers, LiquidityOperations {
+    using FixedPointMathLib for uint256;
+    using CurrencyLibrary for Currency;
+    using LiquidityRangeIdLibrary for LiquidityRange;
+    using PoolIdLibrary for PoolKey;
+    using SafeCast for uint256;
+    using Planner for Planner.Plan;
+    using StateLibrary for IPoolManager;
+
+    PoolId poolId;
+    address alice = makeAddr("ALICE");
+    address bob = makeAddr("BOB");
+
+    uint256 constant STARTING_USER_BALANCE = 10_000_000 ether;
+
+    // expresses the fee as a wad (i.e. 3000 = 0.003e18 = 0.30%)
+    uint256 FEE_WAD;
+
+    LiquidityRange range;
+
+    function setUp() public {
```

Comment:
the setup function for quite a few of these test contracts is the same - deploying routers, currencies, initializing a pool, deploy new posm, do loads of token approvals for alice and bob, set `FEE_WAD`, define `alice` and `bob`. etc
Feels like there should be a `PosmTestSetup` contract with a function to do all of that. Like how we have `deployFreshManagerAndRouters` but something similar for the posm test setup with shared global vars etc

Reply:
+1